lib/Dialect/LLHD/IR/LLHDOps.cpp

```diff
@@ -458,6 +518,29 @@ OpFoldResult llhd::ExtractSliceOp::fold(ArrayRef<Attribute> operands) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// DrvOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult llhd::DrvOp::fold(ArrayRef<Attribute> operands,
+                                SmallVectorImpl<OpFoldResult> &result) {
+  if (!enable())
+    return failure();
+
+  if (matchPattern(enable(), m_Zero())) {
+    getOperation()->dropAllReferences();
```

Comment:
drvop has results right?  If not, you don't need `dropAllReferences`.  If it does, you can't just drop them - you have to replace them with some value.

Either way, I'd recommend writing a testcase for a zero-enable driveop that has a user of the returned value.  In your tests, the results are all dead.

Reply:
`DrvOp` does not have any results. It can be seen as a scheduled/timed `store` with an optional enable argument.
My understanding of `dropAllReferences` was that it removes the operation from the use-list of the operands it uses and is always necessary to be called. Is this done automatically in the case where there are no cycles in the dependences then?

There is already a testcase for a zero-enabled drive, but as drv does not return a value it cannot be used. But I also added one with no enable argument.

Reply:
Ah ok.  Then I think you can just eliminate dropAllReferences.  They are automatically dropped when the operation is deleted.  Please give it a try :-)